server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateCleanupManager.java

```diff
@@ -74,8 +75,13 @@ public class RemoteClusterStateCleanupManager implements Closeable {
     private long lastCleanupAttemptStateVersion;
     private final ThreadPool threadpool;
     private final ClusterApplierService clusterApplierService;
+    private final RemoteRoutingTableService remoteRoutingTableService;
 
-    public RemoteClusterStateCleanupManager(RemoteClusterStateService remoteClusterStateService, ClusterService clusterService) {
+    public RemoteClusterStateCleanupManager(
+        RemoteClusterStateService remoteClusterStateService,
+        ClusterService clusterService,
+        RemoteRoutingTableService remoteRoutingTableService
```

Comment:
Lets pass `Optional<RemoteRoutingTableService>` here similar to how we are using in RemoteClusterStateService